### ISSUE TITLE
Documentation: HttpConnector requires the "tcp" feature

### ISSUE THREAD

**Issue by tux3** [hyperium/hyper#2374]
The docs.rs [documentation for HttpConnector](https://docs.rs/hyper/0.14.1/hyper/client/struct.HttpConnector.html) with hyper 0.14.1 says:

>This is supported on crate feature client and (crate features http1 or http2) only.

As I understand it this seems based on of the following in hyper's lib.rs:

```
cfg_feature! {
    #![all(feature = "client", any(feature = "http1", feature = "http2"))]

    pub mod client;
    pub use crate::client::Client;
}
```

But it seems the HttpConnector struct [also depends on the `tcp` feature](https://github.com/hyperium/hyper/commit/5b348b821c3f43d8dd71179862190932fcca6a1c#diff-1b84118fef4e05f6f4c73c9ad3c5d469ec065f0ef62e20448683dadf6f281c6dL78-R78).

```
#[cfg(feature = "tcp")]
pub use self::connect::HttpConnector;
```

I'm not sure how much control you have over this, but would it be possible to fix the doc to show the extra required feature? =)

**Comment by seanmonstar** [comment]
True, this could be fixed by changing it to use the `cfg_feature` macro instead, like:

```rust
cfg_feature {
    #![feature = "tcp"]
    pub use self::connect::HttpConnector;
}
```

**Comment by SkamDart** [comment]
Looks like I need to fix the commit message but what is the right way to test this? 
I can't quite figure out the incantation that docs.rs uses to generate docs...

**Comment by seanmonstar** [comment]
Fixed in #2377
